File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/KotlinIosPlatformConfig.java
package ru.pocketbyte.locolaser.platform.kotlinmobile;

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.KotlinIosResources;
import ru.pocketbyte.locolaser.resource.PlatformResources;

public class KotlinIosPlatformConfig extends KotlinBaseImplPlatformConfig {

    public static final String TYPE = "kotlin-ios";

    @Override
    public String getType() {
        return TYPE;
    }

    @Override
    protected String getDefaultResourceName() {
        return "ru.pocketbyte.locolaser.StrIos";
    }

    @Override
    public PlatformResources getResources() {
        return new KotlinIosResources(getResourcesDir(), getResourceName(), getInterfaceName());
    }
}

File: platform-kotlin-mobile/src/main/java/ru/pocketbyte/locolaser/platform/kotlinmobile/KotlinIosPlatformConfig.kt
package ru.pocketbyte.locolaser.platform.kotlinmobile

import ru.pocketbyte.locolaser.platform.kotlinmobile.resource.KotlinIosResources

class KotlinIosPlatformConfig : KotlinBaseImplPlatformConfig() {

    companion object {
        const val TYPE = "kotlin-ios"
    }

    override val type = TYPE

    override val defaultResourceName = "ru.pocketbyte.locolaser.StrIos"

    override val resources = KotlinIosResources(resourcesDir!!, resourceName!!, interfaceName!!)

}